### PR TITLE
udev rule to access HID devices on Linux & friends

### PR DESCRIPTION
**Pull request by Qbicz**
Configuration channel script opens the device, on Windows natively and on Linux with hidraw or libusb. On Linux, a proper udev rule is needed for user to do that.

Included also a not-connected fix in the optical sensor.

### CODE REVIEW

**Comment by Qbicz** [comment]
@pdunaj , @b-gent can you review?

File: scripts/hid_configurator/README.rst [hunk 1]
```diff
@@ -38,11 +46,17 @@ On Linux:
 	pip3 install --user -r requirements.txt
 
 This installs:
+
 * HIDAPI library - on Windows provided by `hidapi.dll` or `hidapi-x64.dll`
 * pyhidapi Python wrapper - installed from GitHub repository with additional patches included. The version on PyPI is obsolete and will not work.
 
+On Linux, if you want to be able to call Python script without root rights,
```
Comment (b-gent) [inline scripts/hid_configurator/README.rst]:
```suggestion
On Linux, to call the Python script without root rights,
```

File: scripts/hid_configurator/README.rst [hunk 2]
```diff
@@ -38,11 +46,17 @@ On Linux:
...
+install the provided udev rule `99-hid.rules` by copying it to
+`/etc/udev/rules.d` and re-plugging the device.
```
Comment (b-gent) [inline scripts/hid_configurator/README.rst]:
```suggestion
`/etc/udev/rules.d` and replugging the device.
```

File: scripts/hid_configurator/99-hid.rules [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# root permissions on Linux. Adapted from HIDAPI project.
+
+# To apply the rule, drop this file into /etc/udev/rules.d and unplug
+# and re-plug your device. This is all that is necessary to see the new
+# permissions. Udev does not have to be restarted.
+
+# Rules change the permissions to 0666 (world readable/writable) for specified
+# device on Linux systems.
+
+# Note that for kernels before 2.6.24, you will need
```
Comment (b-gent) [inline scripts/hid_configurator/99-hid.rules]:
```suggestion
# Note that for kernels before 2.6.24, you must
```

File: scripts/hid_configurator/99-hid.rules [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# to substitute "usb" with "usb_device". It shouldn't hurt to use two lines
```
Comment (b-gent) [inline scripts/hid_configurator/99-hid.rules]:
```suggestion
# substitute "usb" with "usb_device". It is recommended to use two lines

File: scripts/hid_configurator/99-hid.rules [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# udev rules which allow using configuration channel for nRF52 Desktop without
+# root permissions on Linux. Adapted from HIDAPI project.
+
+# To apply the rule, drop this file into /etc/udev/rules.d and unplug
```
Comment (b-gent) [inline scripts/hid_configurator/99-hid.rules]:
```suggestion
# To apply the rule, drop this file into `/etc/udev/rules.d` and unplug
```

File: scripts/hid_configurator/99-hid.rules [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+# and re-plug your device. This is all that is necessary to see the new
```
Comment (b-gent) [inline scripts/hid_configurator/99-hid.rules]:
```suggestion
# and replug your device. These are the only steps necessary to see the new
```

File: scripts/hid_configurator/99-hid.rules [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+# permissions. Udev does not have to be restarted.
```
Comment (b-gent) [inline scripts/hid_configurator/99-hid.rules]:
```suggestion
# permissions. You do not need to restart udev.
```

**Comment by Qbicz** [comment]
Tried out the "commit suggestions" feature. Looks like we needed sign-off from both committers. CI is now passing, can you merge @b-gent?

**Pull request merged**